components/infobox/wikis/criticalops/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,88 @@
+---
+-- @Liquipedia
+-- wiki=criticalops
+-- page=Module:Infobox/Person/Player/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local Lua = require('Module:Lua')
+local TeamHistoryAuto = require('Module:TeamHistoryAuto')
+
+local Injector = Lua.import('Module:Infobox/Widget/Injector')
+local Player = Lua.import('Module:Infobox/Person')
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+
+local ROLES = {
+	-- Staff and Talents
+	['analyst'] = {category = 'Analysts', variable = 'Analyst', staff = true},
+	['broadcast analyst'] = {category = 'Broadcast Analysts', variable = 'Broadcast Analyst', talent = true},
+	['caster'] = {category = 'Casters', variable = 'Caster', staff = true},
+}
+
+---@class COPSInfoboxPlayer: Person
+---@field role {category: string, variable: string, staff: boolean?, talent: boolean?}?
+---@field role2 {category: string, variable: string, staff: boolean?, talent: boolean?}?
+local CustomPlayer = Class.new(Player)
+local CustomInjector = Class.new(Injector)
+
+---@param frame Frame
+---@return Html
+function CustomPlayer.run(frame)
+	local player = CustomPlayer(frame)
+	player:setWidgetInjector(CustomInjector(player))
+
+	player.args.history = TeamHistoryAuto.results{convertrole = true}
+	player.role = player:_getRoleData(player.args.role)
+	player.role2 = player:_getRoleData(player.args.role2)
+
+	return player:createInfobox()
+end
+
+---@param id string
+---@param widgets Widget[]
+---@return Widget[]
+function CustomInjector:parse(id, widgets)
+	local caller = self.caller
+
+	if id == 'role' then
+		return {
+			Cell{name = 'Role', content = {caller.role, caller.role2}},
```

Comment:
```suggestion
			Cell{name = 'Role', content = {caller.role.name, caller.role2.name}},
```

Reply:
I have add this line on newest revision, the role doesn't display anymore, but if I keep it unchanged it does showed with unlinked, but will return nil error if nothing was set